### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by LoginALee**

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    name { Faker::Name.first_name }
+    user_name { Faker::Internet.username }
+
+    factory :user_with_conversations do
+      transient do
+        conversations_count { 5 }
+      end
+
+      after(:create) do |user, evaluator|
+        # create(:room, user_id: user.id, conversation_id: create(:conversation).id)
```
Comment (OmauryMarmolejo) [inline spec/factories/users.rb]: Do we need this line?

File: spec/factories/users.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        create_list(:room, evaluator.conversations_count, user_id: user.id) do |room, index|
+        room.conversation_id = create(:conversation).id
+        end
+        user.reload
+      end
+    end
+
+    factory :user_with_messages do
+      transient do
```
Comment (OmauryMarmolejo) [inline spec/factories/users.rb:22]: Cool!

File: spec/factories/users.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        messages_count { 5 }
+      end
+
+      after(:create) do |user, evaluator|
+        create_list(:message, evaluator.messages_count, user: user)
+        user.reload
```
Comment (OmauryMarmolejo) [inline spec/factories/users.rb:28]: Do we need this reload?

**Review by OmauryMarmolejo — COMMENTED** [review]
Cool Alejandro! Take a look at the comments and try to increase the test coverage!

**Comment by ese-varo** [comment]
Great work @LoginALee those factories rock!  :rocket:

**Review by ese-varo — APPROVED** [review]

**Review by jvargas98 — COMMENTED** [review]
Good job! excellent work with your factories, I really liked how you implemented them, just address Omaury's comments.

File: spec/factories/users.rb (hunk 1 above)
Comment (LoginALee) [reply spec/factories/users.rb]: Not really

File: spec/factories/users.rb (hunk 3 above)
Comment (LoginALee) [reply spec/factories/users.rb:28]: I think i read it from the everyday rails book

File: spec/models/message_spec.rb [hunk 4]
```diff
@@ -1,38 +1,31 @@
 require 'rails_helper'
 
-describe Message do
+RSpec.describe Message, type: :model do
```
Comment (ese-varo) [inline spec/models/message_spec.rb]: An RSpec file in the `spec/models` directory is automatically of `type: :model`.
Rspec can be configured to infer the spec type by the location of the faile, you can implement this using: 
```
​# spec/rails_helper.rb
RSpec.configure do |config|
  config.infer_spec_type_from_file_location!
end
```
You can take a look at [the documentation](https://relishapp.com/rspec/rspec-rails/docs/directory-structure)

Comment (LoginALee) [reply spec/models/message_spec.rb]: Yeah, actually i had it that way, but i think i added it trying to fix some error (at the end it was not releated) but forgot to change it, thanks Alvaro! :raised_hands:

**Review by GregVaz — APPROVED** [review]
Good job! Excellent work with your factories and test implementation

File: spec/controllers/conversations_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    end
+
+    it "renders the :show template" do
+      user = create(:user_with_conversations)
+      sign_in user
+      conversation = user.conversations.first
+      get :show, params: { id: conversation }
+      expect(response).to render_template :show
+    end
+  end
```
Comment (GregVaz) [inline spec/controllers/conversations_controller_spec.rb]: You could implement a `before :each` as you already implement it in messages_controller_spec.rb

Comment (LoginALee) [reply spec/controllers/conversations_controller_spec.rb]: Thanks Goyo! i changed them for let (create...) :raised_hands:

File: app/controllers/conversations_controller.rb [hunk 6]
```diff
@@ -6,7 +6,8 @@ def index
     @conversations = ConversationsQuery.new(current_user).of_current_user.by_date
   end
 
-  def show; end
+  def show
```
Comment (OmauryMarmolejo) [inline app/controllers/conversations_controller.rb:9]: Cool!

File: spec/controllers/conversations_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,160 @@
+require 'rails_helper'
+
+RSpec.describe ConversationsController do
+  let(:valid_session) { {} }
```
Comment (OmauryMarmolejo) [inline spec/controllers/conversations_controller_spec.rb]: Do you really need to initialize this here?

Comment (LoginALee) [reply spec/controllers/conversations_controller_spec.rb]: Not really, thanks!

**Pull request merged**